model.py

```diff
@@ -25,66 +25,57 @@ class Genre(db.Model):
 
     __tablename__ = "genres"
 
-    genre_id = db.Column(db.Integer,
-                        autoincrement=True,
-                        primary_key=True)
+    genre_id = db.Column(db.Integer, primary_key=True)
     name = db.Column(db.String(64), nullable=True)
 
     def __repr__(self):
         """Provide helpful representation when printed."""
 
         return f"<Genre genre_id={self.genre_id} name={self.name}>"
 
-# association_table = db.Table('association', 
-#     db.Column('location_id', db.Integer, db.ForeignKey('locations.location_id'), primary_key=True),
-#     db.Column('movie_id', db.Integer, db.ForeignKey('movies.movie_id'),primary_key=True),  
-# )
-#maybe explore this more later
+
+association_table = db.Table('movies_locations',
+                             db.Column('location_id', db.Integer, db.ForeignKey('locations.location_id')),
+                             db.Column('movie_id', db.Integer, db.ForeignKey('movies.movie_id')),
+                             )
+
 
 class Movie(db.Model):
     """Movie on ratings website."""
 
     __tablename__ = "movies"
 
-    movie_id = db.Column(db.Integer,
-                         autoincrement=True,
-                         primary_key=True)
+    movie_id = db.Column(db.Integer, primary_key=True)
     title = db.Column(db.String(100))
     genre_id = db.Column(db.Integer, db.ForeignKey('genres.genre_id'), nullable=False)
-
-    
     genre = db.relationship('Genre', backref='movies')
-    location_ids = db.Column(ARRAY(db.Integer))
-    # locations = db.relationship('Location', 
-    #     secondary=association_table, lazy='subquery',
-    #     backref=db.backref('movies', lazy=True))
-
 
+    # To add locations: movie.locations.append(location); then save as usual using db.session.add/commit
```

Comment:
Note that you'll use this the same way you were trying to do before, but now you don't have to loop through all your movies to find their locations. You simply do this:
```
movie = Movie.query.filter_by(genre=genre)

# Access locations
movie.locations

# add a location
movie.locations.append(location)
db.session.add(movie)
db.session.commit()
```